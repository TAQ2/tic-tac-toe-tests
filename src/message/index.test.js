import React from "react";
import { shallow } from "enzyme";

import Message from "./";

it("message should change depending on whether the game is active or not", () => {
  // @Incomplete - how to check that the two snapshots are not identical
  const wrapper = shallow(<Message isGameActive={true} />);
  expect(wrapper).toMatchSnapshot();

  const wrapper2 = shallow(<Message isGameActive={false} />);
  expect(wrapper2).toMatchSnapshot();
});
