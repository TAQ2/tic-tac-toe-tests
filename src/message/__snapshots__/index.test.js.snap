// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`message should change depending on whether the game is active or not 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Message
    isGameActive={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "",
    },
    "ref": null,
    "rendered": "",
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "",
      },
      "ref": null,
      "rendered": "",
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`message should change depending on whether the game is active or not 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Message
    isGameActive={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Do you wanna play?",
    },
    "ref": null,
    "rendered": "Do you wanna play?",
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Do you wanna play?",
      },
      "ref": null,
      "rendered": "Do you wanna play?",
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
